packages/flutter_driver/lib/src/driver/driver.dart

```diff
@@ -409,10 +409,70 @@ class FlutterDriver {
 
   /// Scrolls the Scrollable ancestor of the widget located by [finder]
   /// until the widget is completely visible.
+  ///
+  /// If the widget located by [finder] is contained by a scrolling widget
+  /// that lazily creates its children, like [ListView] or [CustomScrollView],
+  /// then this method may fail because [finder] doesn't actually exist.
+  /// The [scrollUntilVisible] method can be used in this case.
   Future<Null> scrollIntoView(SerializableFinder finder, { double alignment: 0.0, Duration timeout }) async {
     return await _sendCommand(new ScrollIntoView(finder, alignment: alignment, timeout: timeout)).then((Map<String, dynamic> _) => null);
   }
 
+  /// If widget located by [item] is not visible then repeatedly [scroll] the
+  /// widget located by [scroller] by [dxScroll] and [dyScroll] until [item]
+  /// is visible, and then use [scrollIntoView] to ensure the item's final
+  /// position matches [alignment].
+  ///
+  /// The [scroller] must locate the scrolling widget that contains [item].
+  /// Typically `find.byType('ListView') or `find.byType('CustomScrollView')`.
+  ///
+  /// If [item] is below the currently visible items, then specify a negative
+  /// value for [dyScroll] that's a small enough increment to expose [item]
+  /// without potentially scrolling it up and completely out of view. Similarly
+  /// if [item] is above, then specify a positve value for [dyScroll].
+  ///
+  /// If [item] is to the right of the the currently visible items, then
+  /// specify a negative value for [dxScroll] that's a small enough increment to
+  /// expose [item] without potentially scrolling it up and completely out of
+  /// view. Similarly if [item] is to the left, then specify a positve value
+  /// for [dyScroll].
+  ///
+  /// The [timeout] value should be long enough to accomodate as many scrolls
+  /// as needed to bring an item into view. The default is 10 seconds.
+  Future<Null> scrollUntilVisible (SerializableFinder scroller, SerializableFinder item, {
+    double alignment: 0.0,
+    double dxScroll: 0.0,
+    double dyScroll: -48.0,
```

Comment:
You may not need to have the user specify these values if you implement the logic for this method in `extension.dart`. There you will have access to `RenderObject`s and their dimensions, and will be able to compute optimal `dx`/`dy` values. All we need to specify here is `AxisDirection`.

Reply:
Currently the scrollable is just a widget that contains the widget that will respond to scroll gestures. Locating the actual parent of the children that can be scrolled and inferring how they'll behave under scrolling (will they move at all? maybe just change size?) is a taller order than I think this PR warrants.

Reply:
Ok, if we expect the user to know how much they want to scroll, how about we default `dyScroll` to 0.0? It's strange that we pick a random number for Y but not X. If I want to scroll along the X axis, I will have to zero out `dyScroll` explicitly in the test script, but not when I want to scroll along Y.

Reply:
Good point, I'll change that.